src/tiledb/cloud/geospatial/ingestion.py

```diff
@@ -0,0 +1,1287 @@
+                config = tiledb.Config({"sm.vacuum.mode": mode})
+                try:
+                    tiledb.vacuum(dataset_uri, config=config)
+                except Exception as e:
+                    print(e)
+
+    logger.info("max memory usage: %.3f GiB", max_memory_usage() / (1 << 30))
+
+
+def ingest_datasets(
+    dataset_uri: str,
+    *,
+    dataset_type: DatasetType,
+    acn: Optional[str] = None,
+    config=None,
+    namespace: Optional[str] = None,
+    register_name: Optional[str] = None,
+    search_uri: Optional[str] = None,
+    pattern: Optional[str] = None,
+    ignore: Optional[str] = None,
+    dataset_list_uri: Optional[str] = None,
+    max_files: Optional[int] = None,
+    compression_filter: Optional[dict] = None,
+    workers: int = MAX_WORKERS,
+    batch_size: int = BATCH_SIZE,
+    tile_size: int = RASTER_TILE_SIZE,
+    chunk_size: int = POINT_CLOUD_CHUNK_SIZE,
+    nodata: Optional[float] = None,
+    res: Tuple[float, float] = None,
+    verbose: bool = False,
+    trace: bool = False,
+    log_uri: Optional[str] = None,
+) -> None:
+    """
+    Ingest samples into a dataset.
+
+    :param dataset_uri: dataset URI
+    :param dataset_type: dataset type, one of pointcloud, raster or geometry
+    :param acn: Access Credentials Name (ACN) registered in TileDB Cloud (ARN type),
+        defaults to None
+    :param config: config dictionary, defaults to None
+    :param namespace: TileDB-Cloud namespace, defaults to None
+    :param register_name: name to register the dataset with on TileDB Cloud,
+        defaults to None
+    :param search_uri: URI to search for geospatial dataset files, defaults to None
+    :param pattern: Unix shell style pattern to match when searching for files,
+        defaults to None
+    :param ignore: Unix shell style pattern to ignore when searching for files,
+        defaults to None
+    :param dataset_list_uri: URI with a list of dataset URIs, defaults to None
+    :param max_files: maximum number of URIs to read/find,
+        defaults to None (no limit)
+    :param max_samples: maximum number of samples to ingest, defaults to None (no limit)
+    :param compression_filter: serialized tiledb filter,
+        defaults to None
+    :param workers: number of workers for dataset ingestion, defaults to MAX_WORKERS
+    :param batch_size: batch size for dataset ingestion, defaults to BATCH_SIZE
+    :param tile_size: for rasters this is the tile (block) size
+        for the merged destination array defaults to 1024
+    :param chunk_size: for point cloud this is the PDAL chunk size, defaults to 1000000
+    :param nodata: NODATA value for rasters
+    :param res: Tuple[float, float], output resolution in x/y
+    :param verbose: verbose logging, defaults to False
+    :param trace: bool, enable trace for logging, defaults to False
+    :param log_uri: log array URI
+    """
+
+    # Validate user input
+    if bool(search_uri) & bool(dataset_list_uri):
```

Comment:
Should this be `if bool(search_uri) != bool(dataset_list_uri)`? Otherwise, this will work fine if neither is provided.

Reply:
this is checking that not both `search_uri` and `dataset_list_uri` are provided. I will simplify this.